Core/src/main/java/org/tribuo/onnx/ONNXRef.java

```diff
@@ -0,0 +1,83 @@
+package org.tribuo.onnx;
+
+import ai.onnx.proto.OnnxMl;
+import com.google.protobuf.GeneratedMessageV3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public abstract class ONNXRef<T extends GeneratedMessageV3> {
+    // Unfortunately there is no other shared supertype for OnnxML protobufs
+    protected T backRef;
+    private final String baseName;
+    protected final ONNXContext onnx;
+
+
+    ONNXRef(ONNXContext onnx, T backRef, String baseName) {
+        this.onnx = onnx;
+        this.backRef = backRef;
+        this.baseName = baseName;
+    }
+
+    public abstract String getReference();
+
+    public String getBaseName() {
+        return baseName;
+    }
+
+    public ONNXContext onnx() {
```

Comment:
I like `onnx` here because while there are a lot of ONNXs (ONNXen?) flying about here, in the code that _calls_ this method an instance of `ONNXContext` represents the gateway/intermediary with the ONNX ecosystem. `context` is a very vague name, and there might be other things that are called `context` in a `LinearSGDModel`, but there probably aren't other things called `onnx`.

Reply:
Well, I think that as a method on an `ONNXRef` then something called `onnx()` doesn't really help much. Would `onnxContext()` work?